ntp-udp/src/lib.rs

```diff
@@ -38,8 +41,33 @@ impl UdpSocket {
         loop {
             let mut guard = self.io.writable().await?;
             match guard.try_io(|inner| inner.get_ref().send(buf)) {
-                Ok(result) => return result,
-                Err(_would_block) => continue,
+                Ok(result) => {
+                    if log::log_enabled!(log::Level::Debug) {
+                        if let Err(e) = &result {
+                            debug!("error sending data: {}", e);
+                        }
+                    }
+
+                    if log::log_enabled!(log::Level::Trace) {
+                        if let Ok(size) = &result {
+                            trace!(
+                                "sent {} bytes from {:?} to peer {:?}",
+                                size,
+                                self.as_ref().local_addr(),
+                                self.as_ref().peer_addr()
+                            );
+                        }
+                    }
```

Comment:
we can do a proper match here, dead-code elimination will just remove that if the log level is such that nothing happens in a branch

```rust
match &result {
    Err(e) => { 
        debug!("error sending data: {}", e),
    }
    Ok(size) => {   
        trace!(
            "sent {} bytes from {:?} to peer {:?}",
            size,
            self.as_ref().local_addr(),
            self.as_ref().peer_addr()
        );
    }
}
```

Reply:
Are you sure? The log macros shouldn't be removed as dead code, because the log level is determined at runtime